src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -437,12 +446,13 @@ public static Object createInstance1(
           cause);
     }
 
-    if (result == null && !Modifier.isPublic(declaringClass.getModifiers())) {
-      // result should not be null
-      throw new TestNGException(
-          "An error occurred while instantiating class "
-              + declaringClass.getName()
-              + ". Check to make sure it can be accessed/instantiated.");
+    if (result == null && create) {
+      String suffix = "instantiated";
+      if (!Modifier.isPublic(declaringClass.getModifiers())) {
```

Comment:
I think this is an existing bug, but we are checking if the class is public instead of checking if the desired constructor is `public`. So the message `/accessed` is never going to show up unless and until we are dealing with a non public class.

Reply:
I don't know what was the intention but the constructor may not exist.

Maybe you can have a look at it in another PR?

Reply:
(y) Ok!